File: Adjacency List and Matrix/Adjacency List and Matrix.py
def listt(u, v, listed):
    listed[u - 1].append(v)
    listed[v - 1].append(u)

N, M = map(int, input().split())
matrix = [[0] * N for _ in range(N)]
listed = [[] for _ in range(N)]
for i in range(M):
    u, v = map(int, input().split())
    matrix[u - 1][v - 1] = 1
    matrix[v - 1][u - 1] = 1
    listt(u, v, listed)
for i in range(N):
    listed[i].sort()
    for j in range(len(listed[i])):
        print(listed[i][j], end = " ")
    print()
for i in range(N):
    for j in range(N):
        print(matrix[i][j], end = " ")
    print()

File: Adjacency List and Matrix/Adjacency List and Matrix.cpp
#include <bits/stdc++.h>

using namespace std;
void listt(long long u, long long v, vector<vector<long long>> &listed){
    listed[u - 1].push_back(v);
    listed[v - 1].push_back(u);
}
int main() {
    long long N, M;
    cin >> N >> M;
    long long u, v;
    vector<vector<long long>> matrix(N, vector<long long> (N, 0));
    long long i, j;
    vector<vector<long long>> listed(N);
    for(i = 0; i < M; i++){
        cin >> u >> v;
        matrix[u - 1][v - 1] = 1;
        matrix[v - 1][u - 1] = 1;
        listt(u, v, listed);
    }
    for(i = 0; i < N; i++){
        sort(listed[i].begin(), listed[i].end());
        for(j = 0; j < listed[i].size(); j++){
            cout << listed[i][j] << " ";
        }
        cout << endl;
    }
    for(i = 0; i < N; i++){
        for(j = 0; j < N; j++){
            cout << matrix[i][j] << " ";
        }
        cout << endl;
    }
	return 0;
}
